### ISSUE TITLE
DSL-2 Named output channels

### ISSUE THREAD

**Issue by madkinsz** [nextflow-io/nextflow#1181]
## Named output channels in DSL-2

Hi! I'd like if, instead of accessing output channels from import processes in DSL2 as a list, the output channels could be accessed by a map.

## Usage scenario 

You have a process with multiple output channels e.g.

```
    output:
    set val(sample_name), file('*_trimmed.fastq')
    set val('trimmed'), file('*_trimmed.fastq')
    file '*.log'
```

Accessing these as `process_x[0], [1], [2]..` is not clear what is being accessed.

Alternatively
```
    output:
    set val(sample_name), file('*_trimmed.fastq') as named_fastq
    set val('trimmed'), file('*_trimmed.fastq') as flagged_fastq
    file '*.log' as logs
```

Accessing these as `process_x['named_fastq'], process_x['logs']`

## Suggest implementation 

- Store all outputs in a list as before
- Also store all named outputs in a map
- Add a keyword to define the name of an output (perhaps reuse `into` or use `as`)

**Comment by pditommaso** [comment]
Yes, I was brainstorming on something similar. Like the idea of using `as`.

**Comment by pditommaso** [comment]
Closed in favour of #1290.
